### DISCUSSION TITLE
Testing support?

### DISCUSSION THREAD

**Discussion by tomkcook** [astral-sh/rye#107] [Q&A] [answered]
The documentation has an example `pyproject.toml` which includes a task called `test` and sections called things like `tool.rye."task.pytest"` which look promising as ways of integrating automated tests into a rye workflow - but it would be useful, for those of us who are new to Python packaging, if the documentation spelled out in a little bit more detail exactly how these work.

I, for one, can't figure out how I'm supposed to run `pytest` given the example TOML file.  Can someone give me a pointer please?

**Comment by tomkcook** [comment]
Digging into the source code very slightly, the string "task" doesn't appear anywhere in it.  So I'm mystified how options in pyproject.toml such as `tool.rye."task.pytest"` are meant to work or what the `default_tasks` option is meant to do.

**Reply by mitsuhiko** [reply]
@tomkcook where do you see `task` in the documentation?

**Reply by tomkcook** [reply]
See eg the pyproject.toml [here](https://rye.readthedocs.io/en/latest/) - lots of `tool.rye."task.pytest"` and so on.

**Reply by mitsuhiko** [reply, marked as answer]
@tomkcook that's a completely different project. I was unaware of it's existence. The documentation for this tool are here: https://rye-up.com/guide/pyproject/#toolryescripts

**Reply by tomkcook** [reply]
Well that would explain it.  Thanks for your time.
